docs/performance.md

```diff
@@ -1,7 +1,543 @@
+
+### Warm cache {: #warm_cache }
+
+Finally, we also consider a "**warm cache**" situation, in particular when evaluating start-up time
+for software provided via CernVM-FS. In this situation, all necessary files are available in the
+CernVM-FS client cache (which is typically on the local disk of the client),
+but not yet in the in-memory kernel file system cache. As a result, there will be an impact on
+start-up performance, since the files have to be loaded from (local) disk first.
+
+This is a fairly likely scenario, especially if the CernVM-FS client cache if sufficiently large
+to cover typical workloads being run &ndash; whether that's feasible or not depends a lot on the workload mix.
+
+To evaluate start-up performance in this scenario, we:
+
+1. First do a couple of warmup runs to populate the CernVM-FS client cache;
+2. Clear the kernel file system cache;
+3. Evaluate the start-up performance *once*;
+
+Steps 2 and 3 are then repeated to get multiple performance results in the same scenario.
+
+### Real-world scenario
+
+In a production setup of an HPC cluster, where typically multiple thousands of jobs are being run every day,
+the **actual situation will usually be a mix of these 3 idealistic scenarios**,
+since some files may have been accessed
+very recently (for example common libraries from the
+[EESSI compatibility layer](eessi/high-level-design.md#compat_layer), or common dependencies from
+the [EESSI software layer](eessi/high-level-design.md#software_layer)),
+while others may only be in the CernVM-FS client cache, or not cached yet at all (or may have been evicted from
+the cache due to space constraints).
+
+
+## TensorFlow start-up performance
+
+We have evaluated the start-up performance of [TensorFlow](https://www.tensorflow.org)
+across a wide range of system configurations, by timing how long it takes to run the following command:
+
+```{ .bash .copy }
+python -c 'import tensorflow'
+```
+
+Although this is a trivial single-core workload, it yields some interesting insights
+into start-up performance for software provided via CernVM-FS and alternatives.
+
+More details on this workload, in particular an overview of how many files are required to run it,
+are available in the [*Test workload details* section](#tensorflow_details).
+
+
+### Local filesystems
+
+To set the stage, we first evaluate start-up performance using only local filesystems on the client:
+local disk (SSD, `ext4`, `/tmp`) and RAM disk (DDR4 RAM memory, `/dev/shm`).
+
+Using RAM disk for *all* software installations is not exactly a realistic scenario on an HPC cluster,
+because of the wide variety of software applications and libraries that are typically employed,
+and the total amount of storage space that is required to host the necessary software installations.
+Nevertheless, it is worth evaluating start-up performance with a setup like this, if only to have
+a clear view on what the best possible start-up time is that we can achieve.
```

Comment:
..if only to acquire a baseline of the best possible start-up time on this particular system.